pkg/server/container_execsync.go

```diff
@@ -140,40 +131,33 @@ func (c *criContainerdService) execInContainer(ctx context.Context, id string, o
 		}
 	})
 
-	resCh := make(chan execResult, 1)
-	go func() {
-		// Wait will return if context is cancelled.
-		exitCode, err := process.Wait(ctx)
-		resCh <- execResult{
-			exitCode: exitCode,
-			err:      err,
-		}
-		glog.V(2).Infof("Exec process %q exits with exit code %d and error %v", execID, exitCode, err)
-	}()
+	exitCh, err := process.Wait(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to wait for process %v", err)
```

Comment:
Let's include the execID here (reference the other error). We should include it in the error here because caller doesn't know it.

We don't need to include container id or sandbox id, because caller already knows that.